core/object/object.cpp

```diff
@@ -165,6 +165,38 @@ MethodInfo MethodInfo::from_dict(const Dictionary &p_dict) {
 	return mi;
 }
 
+// This was copied from MethodBind::get_hash() so that the compatibility hashes for virtual and non-virtual methods would be the same.
+uint32_t MethodInfo::get_compatibility_hash() const {
```

Comment:
Should we change current usage of `MethodBind::get_hash()` to use `MethodInfo::get_compatibility_hash()` instead, to keep it consistent?

Reply:
We could. However, the two types of hashes don't necessarily _need_ to be consistent, I'd just prefer them to be, since we've already "battle tested" the hashes from `MethodBind::get_hash()`.

Reply:
@akien-mga I tried having `MethodBind::get_hash()` use `MethodInfo::get_compatibility_hash()` and it turned up a handful of unrelated bugs :-)

For example, there were some methods that had registered more default arguments than they have arguments, which is incorrect and should be fixed. Fixing this means removing the extra default arguments and adding a compatibility hash.

Here's all the changes:

https://github.com/dsnopek/godot/commit/68d7150a6c0545f77fa36fa9138ad0e1d14aad07

Given that these changes are kind of unrelated to the original goal of this PR, would you prefer (1) that I save those for a follow-up PR or (2) should I merge those changes into this PR anyway?

Reply:
I think you can save them for a follow-up PR. Looks like a good cleanup!

Reply:
Ok, thanks!

Reply:
I just posted PR https://github.com/godotengine/godot/pull/101449 with these changes